### ISSUE TITLE
Issues with HDDriver 10.0

### ISSUE THREAD

**Issue by miniupnp** [atarijookie/ce-atari#104]
http://www.atari-forum.com/viewtopic.php?f=103&t=30854

**Comment by miniupnp** [comment]
it looks like the problem is HDDriver 10 badly detected ICD extensions.
I've tested from disk image, it works

**Comment by miniupnp** [comment]
below is the log of the boot of HDDriver 10.0 on a IMG mounted as ACSI ID 0
we see the ICD command INQUIRY ($12) and then READ_CAPACITY ($25)
```
02249798 00000002 (2016-12-10 09:35:47.140077)  handleAcsiCommand: 08 00 00 00 01 00
02249800 00000002 (2016-12-10 09:35:47.141798)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x000000 lenX=0x01
02249818 00000002 (2016-12-10 09:35:47.159923)  handleAcsiCommand: 08 00 00 02 01 00
02249820 00000002 (2016-12-10 09:35:47.161663)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x000002 lenX=0x01
02249838 00000002 (2016-12-10 09:35:47.179581)  handleAcsiCommand: 08 00 00 46 01 00
02249839 00000001 (2016-12-10 09:35:47.181143)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x000046 lenX=0x01
02249857 00000001 (2016-12-10 09:35:47.199274)  handleAcsiCommand: 08 00 04 ee 08 00
02249859 00000002 (2016-12-10 09:35:47.200846)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x0004ee lenX=0x08
02249878 00000002 (2016-12-10 09:35:47.219680)  handleAcsiCommand: 08 00 00 06 01 00
02249879 00000001 (2016-12-10 09:35:47.221258)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x000006 lenX=0x01
02249898 00000002 (2016-12-10 09:35:47.239476)  handleAcsiCommand: 08 00 04 f6 08 00
02249899 00000001 (2016-12-10 09:35:47.241058)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x0004f6 lenX=0x08
02249918 00000002 (2016-12-10 09:35:47.259847)  handleAcsiCommand: 08 00 04 fe 08 00
02249920 00000002 (2016-12-10 09:35:47.261607)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x0004fe lenX=0x08
02249943 00000002 (2016-12-10 09:35:47.284751)  handleAcsiCommand: 08 00 05 06 08 00
02249944 00000001 (2016-12-10 09:35:47.286400)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x000506 lenX=0x08
02249962 00000001 (2016-12-10 09:35:47.304247)  handleAcsiCommand: 08 00 05 0e 08 00
02249964 00000002 (2016-12-10 09:35:47.305832)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x00050e lenX=0x08
02249983 00000002 (2016-12-10 09:35:47.324663)  handleAcsiCommand: 08 00 05 16 08 00
02249984 00000001 (2016-12-10 09:35:47.326406)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x000516 lenX=0x08
02250003 00000002 (2016-12-10 09:35:47.345006)  handleAcsiCommand: 08 00 05 1e 08 00
02250005 00000002 (2016-12-10 09:35:47.346584)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x00051e lenX=0x08
02250047 00000001 (2016-12-10 09:35:47.389233)  handleAcsiCommand: 12 00 00 00 10 00
02250054 00000002 (2016-12-10 09:35:47.396042)  handleAcsiCommand: 1f 12 00 00 00 10 00 00 00 00 00 10 00 00 isIcd
02250073 00000002 (2016-12-10 09:35:47.414682)  handleAcsiCommand: 12 00 00 00 20 00
02250094 00000002 (2016-12-10 09:35:47.435466)  handleAcsiCommand: 1f 25 00 00 00 00 00 00 00 00 00 10 00 00 isIcd
02250099 00000002 (2016-12-10 09:35:47.441138)  handleAcsiCommand: 12 00 00 00 0f 00
02250108 00000001 (2016-12-10 09:35:47.450046)  handleAcsiCommand: 08 00 00 00 01 00
02250110 00000002 (2016-12-10 09:35:47.451912)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x000000 lenX=0x01
02250119 00000001 (2016-12-10 09:35:47.461092)  handleAcsiCommand: 08 00 00 02 01 00
02250121 00000002 (2016-12-10 09:35:47.462955)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x000002 lenX=0x01
02250129 00000002 (2016-12-10 09:35:47.471109)  handleAcsiCommand: 08 01 00 01 01 00
02250131 00000002 (2016-12-10 09:35:47.473015)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x010001 lenX=0x01
02250679 00000002 (2016-12-10 09:35:48.020639)  handleAcsiCommand: 72 00 00 00 20 00
02250700 00000002 (2016-12-10 09:35:48.041692)  handleAcsiCommand: 68 00 00 00 01 00
02250701 00000001 (2016-12-10 09:35:48.043401)  Scsi::SCSI_ReadWrite6(READ) ID=3 sector=0x000000 lenX=0x01
02252297 00000002 (2016-12-10 09:35:49.638793)  handleAcsiCommand: 00 00 00 00 00 00
02252346 00000002 (2016-12-10 09:35:49.688034)  handleAcsiCommand: 08 00 00 46 04 00
02252348 00000002 (2016-12-10 09:35:49.689669)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x000046 lenX=0x04
02252359 00000001 (2016-12-10 09:35:49.701164)  handleAcsiCommand: 08 00 05 2e 04 00
02252361 00000002 (2016-12-10 09:35:49.703025)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x00052e lenX=0x04
02252383 00000001 (2016-12-10 09:35:49.725350)  handleAcsiCommand: 08 00 05 36 04 00
02252385 00000002 (2016-12-10 09:35:49.726988)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x000536 lenX=0x04
02252398 00000002 (2016-12-10 09:35:49.739725)  handleAcsiCommand: 08 00 05 3a 04 00
02252399 00000001 (2016-12-10 09:35:49.741313)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x00053a lenX=0x04
02252410 00000002 (2016-12-10 09:35:49.752063)  handleAcsiCommand: 08 00 00 26 04 00
02252412 00000002 (2016-12-10 09:35:49.753669)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x000026 lenX=0x04
02252423 00000001 (2016-12-10 09:35:49.765170)  handleAcsiCommand: 08 00 05 3e 20 00
02252425 00000002 (2016-12-10 09:35:49.766767)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x00053e lenX=0x20
02252448 00000002 (2016-12-10 09:35:49.790258)  handleAcsiCommand: 08 00 05 5e 04 00
02252450 00000002 (2016-12-10 09:35:49.791952)  Scsi::SCSI_ReadWrite6(READ) ID=0 sector=0x00055e lenX=0x04
```

**Comment by miniupnp** [comment]
HDDriver 10 detects the ICD extension on the IMG (ACSI ID 0) but not the SD CARD (ACSI ID 1)
![hddriver_bug](https://cloud.githubusercontent.com/assets/1070377/21072768/d9c736c2-becd-11e6-8f9d-d76ffc7a4a81.jpg)

**Comment by miniupnp** [comment]
In HDDRutil, when I rescan devices, everything is OK !

before rescaning : 
![hddrutil1](https://cloud.githubusercontent.com/assets/1070377/21072811/f512d6b0-bece-11e6-82e3-b06f46fb61a7.png)

after rescaning : 
![hddrutil3](https://cloud.githubusercontent.com/assets/1070377/21072815/ff70be06-bece-11e6-8ff8-17ec7eb240cd.png)

looks like a HDDriver 10.0 bug to me

**Comment by miniupnp** [comment]
it is indeed a problem in HDDriver 10.0. It will be solved in 10.01 : 
```
- HDDRIVER: Fixed a problem in HDDRIVER 10.00 that affected the automatic
  detection of ICD compatible adapters. (10.01)
```
http://www.atari-forum.com/viewtopic.php?p=307241#p307241
